src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -669,6 +669,12 @@ public bool IsInNamespaceOrTypeContext(SyntaxNode node)
             return SyntaxFacts.IsInNamespaceOrTypeContext(node as ExpressionSyntax);
         }
 
+        public bool IsBaseTypeInBaseList(SyntaxNode node)
+        {
+            var baseTypeNode = node as BaseTypeSyntax ?? node?.Parent as BaseTypeSyntax;
+            return baseTypeNode.IsParentKind(SyntaxKind.BaseList);
```

Comment:
not really a syntax fact.  also, implementation doesn't match name.  It's really asking if this is a type that is a direct child (counting the intermediary base-type syntax) of the BaseList.  I would not have this be in Syntaxfacts.

Reply:
I can renamed the method. However, I am still not clear what classifies to be in SyntaxFactsService and what does not. This seems like a very basic syntactic question whose answer is valuable at the core layer.